src/AlgebraicGeometry/Schemes/Covering/Objects/Attributes.jl

```diff
@@ -107,18 +107,22 @@ function inherit_decomposition_info!(
 
   for V in patches(orig_cov)
     # Collect the patches in `ref_cov` refining V
-    V_ref = [U for U in patches(ref_cov) if decomp_dict[U][1] === V]
-    # Collect the corresponding complement equations
-    comp_eqns = [decomp_dict[U][2] for U in V_ref]
+    V_ref = [(U, (UV, h)) for (U, (UV, h)) in decomp_dict if UV === V]
+    _compl(a) = sum(length(lifted_numerator(b)) + length(lifted_denominator(b)) for b in a[2][2]; init=0)
+    V_ref = sort!(V_ref, by=_compl)
     # Start out from the original decomposition info
     dec_inf = copy(decomposition_info(orig_cov)[V])
-    for (i, U) in enumerate(V_ref)
+    for (i, (U, (_, h))) in enumerate(V_ref)
```

Comment:
With my suggested change above, this would be:
```suggestion
    for (i, (U, h)) in enumerate(V_ref)
```